squidpy/_constants/_pkg_constants.py

```diff
@@ -134,6 +138,63 @@ def spot_diameter(
                     f"`adata.uns[{spatial_key!r}][{library_id!r}]['scalefactors'][{spot_diameter_key!r}]]`"
                 ) from None
 
+        @classmethod
+        def library_id(
+            cls,
+            adata: AnnData,
+            spatial_key: str,
+            library_id: Sequence[str] | str | None = None,
+            return_all: bool = False,
+        ) -> Sequence[str] | str | None:
+            library_id = cls._haystack(adata, spatial_key, library_id, sub_key=None)
+            if library_id is not None and not return_all:
+                if len(library_id) > 1:
+                    raise ValueError(
+                        f"Unable to determine which library id to use. "
+                        f"Please specify one from: `{sorted(library_id)}`."
+                    )
+                return library_id[0]
+            if len(library_id) > 0:
+                return library_id
+            return None
+
+        @classmethod
+        def library_mapping(
+            cls,
+            adata: AnnData,
+            spatial_key: str,
+            sub_key: str,
+            library_id: Sequence[str] | str | None = None,
+        ) -> Mapping[str, Sequence[str]]:
+            library_id = cls._haystack(adata, spatial_key, library_id, sub_key)
+            return {i: list(adata.uns[spatial_key][i][sub_key].keys()) for i in library_id}
+
+        @classmethod
+        def _haystack(
+            cls,
+            adata: AnnData,
+            spatial_key: str,
+            library_id: Sequence[str] | str | None = None,
+            sub_key: Optional[str] = None,
+        ) -> Sequence[str]:
+            if spatial_key not in adata.uns:
+                raise KeyError(f"Spatial key {spatial_key!r} not found in `adata.uns`.")
+            haystack = list(adata.uns[spatial_key].keys())
```

Comment:
There are some unnecessary calls to `.keys` (usually, it should suffice that the object implement `__getitem__` only); won't mention elsewhere

Reply:
I see the point but I think it's more understandable